File: collect_app/src/main/java/org/odk/collect/android/formentry/repeats/AddRepeatDialog.java
package org.odk.collect.android.formentry.repeats;

import android.content.Context;
import android.content.DialogInterface;

import androidx.appcompat.app.AlertDialog;

import static android.content.DialogInterface.BUTTON_NEGATIVE;
import static android.content.DialogInterface.BUTTON_POSITIVE;

import com.google.android.material.dialog.MaterialAlertDialogBuilder;

public class AddRepeatDialog {

    private AddRepeatDialog() {
    }

    public static void show(Context context, String groupLabel, Listener listener) {
        AlertDialog alertDialog = new MaterialAlertDialogBuilder(context).create();
        DialogInterface.OnClickListener repeatListener = (dialog, i) -> {
            switch (i) {
                case BUTTON_POSITIVE -> listener.onAddRepeatClicked();
                case BUTTON_NEGATIVE -> listener.onCancelClicked();
            }
        };

        String dialogMessage;
        if (groupLabel.isBlank()) {
            dialogMessage = context.getString(org.odk.collect.strings.R.string.add_another_question);
        } else {
            dialogMessage = context.getString(org.odk.collect.strings.R.string.add_repeat_question,
                    groupLabel);
        }

        alertDialog.setTitle(dialogMessage);

        alertDialog.setButton(BUTTON_POSITIVE, context.getString(org.odk.collect.strings.R.string.add_repeat),
                repeatListener);
        alertDialog.setButton(BUTTON_NEGATIVE, context.getString(org.odk.collect.strings.R.string.cancel),
                repeatListener);

        alertDialog.setCancelable(false);
        alertDialog.show();
    }

    public interface Listener {
        void onAddRepeatClicked();

        void onCancelClicked();
    }
}

File: collect_app/src/main/java/org/odk/collect/android/formentry/repeats/AddRepeatDialog.kt
package org.odk.collect.android.formentry.repeats

import android.content.Context
import android.view.LayoutInflater
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.google.android.material.textview.MaterialTextView
import org.odk.collect.android.R
import org.odk.collect.androidshared.ui.multiclicksafe.MultiClickSafeMaterialButton
import org.odk.collect.strings.R.string

object AddRepeatDialog {

    @JvmStatic
    fun show(context: Context, groupLabel: String?, listener: Listener) {
        val view = LayoutInflater.from(context).inflate(R.layout.add_repeat_dialog_layout, null)

        view.findViewById<MaterialTextView>(R.id.message).text = if (groupLabel.isNullOrBlank()) {
            context.getString(string.add_another_question)
        } else {
            context.getString(string.add_repeat_question, groupLabel)
        }

        val alertDialog = MaterialAlertDialogBuilder(context)
            .setView(view)
            .setCancelable(false)
            .create()

        view.findViewById<MultiClickSafeMaterialButton>(R.id.add_button).setOnClickListener {
            listener.onAddRepeatClicked()
            alertDialog.dismiss()
        }

        view.findViewById<MultiClickSafeMaterialButton>(R.id.do_not_add_button).setOnClickListener {
            listener.onCancelClicked()
            alertDialog.dismiss()
        }

        alertDialog.show()
    }

    interface Listener {
        fun onAddRepeatClicked()

        fun onCancelClicked()
    }
}
